tests/beaker_tests/cisco_upgrade/test_upgrade_idempotence.rb

```diff
@@ -0,0 +1,59 @@
+###############################################################################
+# Copyright (c) 2017 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+@version = image_version.to_s.strip
+# Test hash top-level keys
+tests = {
+  master:           master,
+  agent:            agent,
+  operating_system: 'nexus',
+  ensurable:        false,
+  resource_name:    'cisco_upgrade',
+}
+
+# Non-default Tests. NOTE: [:resource] = [:manifest_props] for all non-default
+tests[:non_default] = {
+  desc:           '1.1 Non_Defaults',
+  code:           [0],
+  title_pattern:  'image',
+  manifest_props: {
+    version:           @version,
+    source_uri:        on(agent, facter_cmd('-p cisco.images.system_image')).output,
```

Comment:
The test would most likely fail because node-utils would return fail. We could skip the test if image not on bootflash: , but in a controlled test environment, I'm not sure if we're going to run into this.